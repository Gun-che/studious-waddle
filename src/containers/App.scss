.App-header {
  background-color: #282c34;
  min-height: 60px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  font-size: calc(10px + 2vmin);
  color: white;
  ul {
    list-style: none;
    display: flex;
    li {
      padding-right: 10px;
    }
  }
  a {
    color: white;
    &.active {
      color: red;
    }
  }
  .logo {
    margin-right: auto;
  }
  .button-box {
    display: flex;
    flex-direction: column;
    button {
      width: 100px;
      border-radius: 50%;
      height: 100px;
      &.login {
        margin-bottom: -60px;
        &:hover {
          z-index: 10;
        }
      }
    }
  }
}
