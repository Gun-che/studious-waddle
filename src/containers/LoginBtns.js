import React from 'react';

export default function LoginBtns(props) {
  return (
    <div className='button-box'>
      <button className='login btn'>Log In</button>
      <button className='registration btn'>Registration</button>
    </div>
  )
}