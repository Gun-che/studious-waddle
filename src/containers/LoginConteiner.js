import React, { useState } from 'react';
import { connect } from 'react-redux';
import { handlerLogin } from '../actions/UserAction';
import PromptLogin from '../components/PromptLogin/PromptLogin'
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';


export function Login(props) {

  const [login, setLogin] = useState('');
  const [pass, setPass] = useState('');

  function handlerSubmit(e) {
    e.preventDefault();
    props.handlerLogin(login, pass)
    setPass('')
  }

  function handlerChange(e) {
    if (e.target.id === 'login') {
      setLogin(e.target.value)
    } else if (e.target.id === 'password') {
      setPass(e.target.value)
    }
  }


  return (
    <>
      {props.loggedIn ? <Redirect to='/profile' /> : <PromptLogin
        handlerChange={handlerChange}
        handlerSubmit={handlerSubmit}
        user={props.user}
        message={props.message}
        success={props.success}
        login={login}
        pass={pass} />}
    </>
  )
}

const mapStateToProps = store => {
  return {
    user: store.user.login,
    message: store.user.message,
    success: store.user.success,
    loggedIn: store.user.loggedIn,
  }
}

const mapDispatchToProps = dispatch => {
  return {
    handlerLogin: (login, pass) => dispatch(handlerLogin(login, pass))
  }
}

Login.propTypes = {
  user: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  success: PropTypes.bool.isRequired,
  loggedIn: PropTypes.bool.isRequired,
  handlerLogin: PropTypes.func.isRequired
}


export default connect(mapStateToProps, mapDispatchToProps)(Login)

