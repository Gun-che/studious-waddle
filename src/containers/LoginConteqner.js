import React, { useState } from 'react';
import { connect } from 'react-redux';
import { handlerLogin } from '../actions/UserAction';
import PromptLogin from '../components/PromptLogin/PromptLogin'
import { Redirect } from 'react-router-dom';


function Login(props) {

  const [login, setLogin] = useState('');
  const [pass, setPass] = useState('');

  function handlerSubmit(e) {
    e.preventDefault();
    props.handlerLogin(login, pass)
  }

  function handlerChange(e) {
    if (e.target.id === 'login') {
      setLogin(e.target.value)
    } else if (e.target.id === 'password') {
      setPass(e.target.value)
    }
  }


  return (
    <>
      {false ? <Redirect to='/exit' /> : <PromptLogin
        handlerChange={handlerChange}
        handlerSubmit={handlerSubmit}
        user={props.user}
        message={props.message}
        success={props.success} />}
    </>
  )
}

const mapStateToProps = store => {
  return {
    user: store.user.login,
    message: store.user.message,
    success: store.user.success,
    loggedIn: store.user.loggedIn,
  }
}

const mapDispatchToProps = dispatch => {
  return {
    handlerLogin: (login, pass) => dispatch(handlerLogin(login, pass))
  }
}


export default connect(mapStateToProps, mapDispatchToProps)(Login)

