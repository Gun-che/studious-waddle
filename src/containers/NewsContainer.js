import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux';
import { handlerNewsRequest } from '../actions/NewsAction';

import News from '../components/News/News';

function NewsContainer(props) {
  return (
    <News
      data={props.data}
      message={props.message}
      isFetching={props.isFetching}
      handlerRequest={props.handlerRequest}
    />
  )
}

NewsContainer.propTypes = {
  data: PropTypes.array.isRequired,
  message: PropTypes.string.isRequired,
  isFetching: PropTypes.bool.isRequired,
  handlerRequest: PropTypes.func.isRequired,
}

const mapStateToProps = store => ({
  data: store.news.data,
  isFetching: store.news.isFetching,
  message: store.news.message,
})

const mapDispatchToProps = dispatch => ({
  handlerRequest: () => dispatch(handlerNewsRequest()),
})

export default connect(mapStateToProps, mapDispatchToProps)(NewsContainer)