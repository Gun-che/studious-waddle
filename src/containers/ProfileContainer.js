import React from 'react';
import PropTypes from 'prop-types'
import { connect } from 'react-redux';
import { handlerUserInfoRequest } from '../actions/ProfileAction';

import Profile from '../components/Profile/Profile';

function ProfileContainer(props) {

  return (
    <Profile
      data={props.data}
      isFetching={props.isFetching}
      message={props.message}
      handlerRequest={props.handlerRequest}
      loggedIn={props.loggedIn}
      login={props.login}
      userId={props.userId}
    />
  )

}

ProfileContainer.propTypes = {
  data: PropTypes.object.isRequired,
  isFetching: PropTypes.bool.isRequired,
  message: PropTypes.string.isRequired,
  handlerRequest: PropTypes.func.isRequired,
  loggedIn: PropTypes.bool.isRequired,
  userId: PropTypes.number.isRequired,
  login: PropTypes.string.isRequired
}

const mapStateToProps = store => {
  return {
    data: store.profile.data,
    isFetching: store.profile.isFetching,
    message: store.profile.message,
    loggedIn: store.user.loggedIn,
    userId: store.user.userId,
    login: store.user.login,
  }
}

const mapDispatchToProps = dispatch => {
  return {
    handlerRequest: id => dispatch(handlerUserInfoRequest(id)),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ProfileContainer)

