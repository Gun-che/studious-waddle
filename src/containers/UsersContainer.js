import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { handlerUsersRequest } from '../actions/PeoplesAction'

import Users from '../components/Users/Users'

export const UserContainer = (props) => {

  return (
    <Users
      isFetching={props.isFetching}
      data={props.data}
      message={props.message}
      handlerRequest={props.handlerRequest}
    />
  )
}

UserContainer.propTypes = {
  isFetching: PropTypes.bool.isRequired,
  data: PropTypes.array.isRequired,
  handlerRequest: PropTypes.func.isRequired,
}

UserContainer.defaultProps = {
  message: null,
}

const mapStateToProps = (store) => ({
  isFetching: store.people.isFetching,
  data: store.people.data,
  message: store.people.message,
})

const mapDispatchToProps = (dispatch) => ({
  handlerRequest: (amount, prevData) => dispatch(handlerUsersRequest(amount, prevData))
})

export default connect(mapStateToProps, mapDispatchToProps)(UserContainer)



