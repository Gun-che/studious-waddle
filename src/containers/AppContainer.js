import React from 'react';
import loadable from '@loadable/component'
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  NavLink,
  Switch,
  BrowserRouter as Router,
  Route,
} from 'react-router-dom';

import { handlerExit } from '../actions/UserAction'
import LoadingComponent from '../components/LoadingComponent/LoadingComponent';
import Exit from '../components/Exit/Exit'
import { ReactComponent as Logo } from '../logo.svg';

import './App.scss';

function App(props) {

  return (
    <Router>
      <div className="App">
        <header className="App-header">
          <div className='logo'>
            <NavLink strict to='/'
              activeClassName='active'
            ><Logo width="70" heigth="70" />logo</NavLink>

          </div>
          <nav>
            <ul>
              <li>
                <NavLink
                  to='/profile'
                  activeClassName='active'
                >Profile</NavLink>
              </li>
              <li>
                <NavLink
                  to='/users'
                  activeClassName='active'>Users</NavLink>
              </li>
              <li>
                <NavLink to='/news'
                  activeClassName='active'>News</NavLink>
              </li>
            </ul>
          </nav>
          <NavLink to='./login'
            activeClassName='active'><button>{props.loggedIn ? 'Exit' : 'LogIn'}</button></NavLink>
        </header>
        <main>
          <section className='main'>
            <Switch>
              <Route exact path='/'>
                <Home></Home>
              </Route>
              <Route exact path='/profile'>
                <Profile loggedIn={props.loggedIn}></Profile>
              </Route>
              <Route exact path='/users'>
                <Users></Users>
              </Route>
              <Route exact path='/news'>
                <News></News>
              </Route>
              <Route exact path='/login'>
                <Login></Login>
              </Route>
              <Route exact path='/exit'>
                <Exit handlerExit={props.handlerExit}></Exit>
              </Route>
            </Switch>
          </section>
        </main>
      </div>
    </Router>
  );
}

export default App;

const Home = loadable(() => import('../components/Home/Home'), {
  fallback: LoadingComponent
})

const Profile = loadable(() => import('../components/Profile/Profile'), {
  fallback: LoadingComponent
})

const Users = loadable(() => import('../components/Users/Users'), {
  fallback: LoadingComponent
})

const News = loadable(() => import('../components/News/News'), {
  fallback: LoadingComponent
})

const Login = loadable(() => import('./LoginConteqner'), {
  fallback: LoadingComponent
})

const mapDispatchToProps = dispatch => {
  return {
    handlerExit: dispatch(handlerExit())
  }
}
connect((store) => ({ loggedIn: store.user.loggedIn }), mapDispatchToProps)