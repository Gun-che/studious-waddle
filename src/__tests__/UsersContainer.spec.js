import React from 'react'
import { shallow, mount } from 'enzyme'
import { UserContainer } from '../containers/UsersContainer'

let res = {
  dob: {
    date: "1947-02-28T21:54:18.936Z",
    age: 73
  },
  email: "dusan.guillot@example.com",
  gender: "male",
  location: {
    street: {
      name: 1,
      number: 2
    },
    city: "Rochefort",
    state: "Solothurn",
    country: "Switzerland",
  },
  login: {
    username: "greenfrog151",
    md5: "5bbf83442b2531c94faff7de1534a884",
  },
  name: {
    title: "Monsieur",
    first: "Dusan",
    last: "Guillot"
  },
  phone: "075 076 26 28",
  picture: {
    large: 1, medium: 2,
  }
}


describe('Users container', () => {
  const props = {
    message: null,
    isFetching: false,
    handlerRequest: () => { },
    data: [],
  }
  describe('Users container init', () => {
    const UsersContainer = shallow(<UserContainer {...props} />)

    it('renders properly', () => {
      expect(UsersContainer).toMatchSnapshot()
    })
  })

  // describe('Users container loading', () => {
  //   const nextProps = {
  //     ...props,
  //     isFetching: true,
  //     data: []
  //   }

  //   const usersContainer = mount(<UserContainer {...nextProps} />)

  //   it('render preloader', () => {
  //     expect(usersContainer.find('Loading')).toHaveLength(1)
  //   })

  //   it('renders properly', () => {
  //     expect(usersContainer).toMatchSnapshot()
  //   })
  // })

  describe('Users container render <Users>', () => {
    const nextProps = {
      ...props,
      data: [res]
    }

    const usersContainer = shallow(<UserContainer {...nextProps} />)

    it('render <Users>', () => {
      expect(usersContainer.find('Users')).toHaveLength(1)
    })

    it('renders properly', () => {
      expect(usersContainer).toMatchSnapshot()
    })
  })

  // describe('Users container with error', () => {
  //   const nextProps = {
  //     ...props,
  //     message: {
  //       status: 'Something going wrong',
  //     },
  //     isFetching: false,

  //   }

  //   const usersContainer = mount(<UserContainer {...nextProps} />)

  //   it('render message', () => {
  //     expect(usersContainer.find('h2').text()).toEqual('Something going wrong ' + nextProps.message.status)
  //   })

  //   it('renders properly', () => {
  //     expect(usersContainer).toMatchSnapshot()
  //   })
  // })

  //   describe('Users container init', () => {

  //     const mockFetchGetUsers = jest.fn()
  //     const nextProps = {
  //       ...props,
  //       handlerRequest: mockFetchGetUsers,
  //     }

  //     const UsersContainer = mount(<UserContainer {...nextProps} />)

  //     it('renders properly', () => {
  //       expect(UsersContainer).toMatchSnapshot()
  //     })

  //     it('dispatches the handlerRequest method it receives from props', () => {

  //       expect(mockFetchGetUsers).toHaveBeenCalled()
  //     })
  //   })
})