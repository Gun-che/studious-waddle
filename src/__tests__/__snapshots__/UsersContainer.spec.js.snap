// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users container Users container init renders properly 1`] = `ShallowWrapper {}`;

exports[`Users container Users container render <Users> renders properly 1`] = `ShallowWrapper {}`;
